packages/netlify-cms-backend-gitlab/src/API.js

```diff
@@ -224,58 +260,443 @@ export default class API {
+      .then(metadata => this.mergeMR(metadata.mr, metadata.branch, metadata.objects))
+      .then(() => this.deleteBranch(branchName))
+      .then(() => this.deleteMetadata(contentKey));
+  }
+
+  // Branches
+
+  generateBranchName(name) {
+    return `${CMS_BRANCH_PREFIX}${name}`;
+  }
+
+  assertBranchName(name) {
+    return name.startsWith(CMS_BRANCH_PREFIX);
+  }
+
+  getBranch(branch = this.branch) {
+    return this.requestJSON(`${this.repoURL}/repository/branches/${encodeURIComponent(branch)}`);
+  }
+
+  createBranch(branch, ref = this.branch) {
+    return this.requestJSON({
+      method: 'POST',
+      url: `${this.repoURL}/repository/branches`,
+      params: { branch, ref },
+    });
+  }
+
+  deleteBranch(branch) {
+    return this.request({
+      method: 'DELETE',
+      url: `${this.repoURL}/repository/branches/${encodeURIComponent(branch)}`,
+    });
+  }
+
+  // Merge Requests
+
+  getMR(mrNumber) {
+    return this.requestJSON(
+      `${this.repoURL}/repository/merge_requests/${encodeURIComponent(mrNumber)}`,
+    );
+  }
+
+  createMR(title, source_branch, target_branch = this.branch) {
+    console.log('%c Creating MR', 'line-height: 30px;text-align: center;font-weight: bold');
+    return this.requestJSON({
+      method: 'POST',
+      url: `${this.repoURL}/merge_requests`,
+      body: {
+        source_branch,
+        target_branch,
+        title,
+        description: 'Generated by Netlify CMS',
+        remove_source_branch: false,
+        squash: this.squash_merges,
+      },
+    });
+  }
+
+  closeMR(mr) {
+    const mrNumber = mr.number;
+    console.log('%c Deleting MR', 'line-height: 30px;text-align: center;font-weight: bold');
+    return this.requestJSON({
+      method: 'PUT',
+      url: `${this.repoURL}/merge_requests/${encodeURIComponent(mrNumber)}`,
+      body: {
+        state_event: 'close',
+      },
+    });
+  }
+
+  mergeMR(mr, branch, objects) {
+    const mrNumber = mr.number;
+    const sha = mr.head;
+    console.log('%c Merging MR', 'line-height: 30px;text-align: center;font-weight: bold');
+    return this.requestJSON({
+      method: 'PUT',
+      url: `${this.repoURL}/merge_requests/${encodeURIComponent(mrNumber)}/merge`,
+      body: {
+        merge_commit_message:
+          'Merged by Netlify CMS\n' + `Merge branch '${branch}' into '${this.branch}'`,
+        sha,
+      },
+    }).catch(error => {
+      if (error instanceof APIError && error.status === 405) {
+        return this.forceMergeMR(mr, branch, objects);
+      }
+      throw error;
+    });
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  forceMergeMR(mr, branch, objects) {
+    // TODO - rudolf - Why is this needed? How are merge conflicts possible?
+    throw new Error(`Not Implemented`);
+  }
+
+  // eslint-disable-next-line no-unused-vars
+  rebaseMR(mrNumber, branch, contentKey, metadata, sha) {
+    // TODO - rudolf - Why is this needed? How are assets affected by branches?
```

Comment:
Depending on the setting for a GitLab repository, it might be only allowed to merge if a fast forward merge is possible.  
That means if someone has merged something while you've created a new merge request you need to rebase in order to merge your own changes.